shared/elements/gaia_lowstorage_dialog/script.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+/* global ComponentUtils, MozActivity */
+
+/**
+ * The gaia-confirm component displays a dialog in which the user has a
+ * choice to confirm or cancel the action. It may be displayed along with a
+ * title, description, and image. Buttons may also be configured.
+ * @requires GaiaButtons
+ */
+window.GaiaLowstorageDialog = (function(win) {
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Allow baseurl to be overridden (used for demo page)
+  var baseurl = window.GaiaLowstorageDialogBaseurl ||
+    '/shared/elements/gaia_lowstorage_dialog/';
+
+  function learnmore() {
+    var activity = new MozActivity({
+      name: 'configure',
+      data: {
+        section: 'applicationStorage'
+      }
+    });
+
+    return Promise.resolve(activity);
+  }
+
+  /**
+   * Localize the component manually as l10n attributes are not supported
+   * within the shadow dom. See also: bug 1026236.
+   */
+  proto._localizeShadowDom = function() {
+    navigator.mozL10n.translateFragment(this.shadowRoot);
+  };
+
+  proto.createdCallback = function() {
+    var shadow = this.createShadowRoot();
+
+    var node = document.importNode(template.content, true);
+    shadow.appendChild(node);
+    ComponentUtils.style.call(this, baseurl);
+
+    navigator.mozL10n.ready(this._localizeShadowDom.bind(this));
```

Comment:
Answering to @azasypkin:

In my mind, they have quite different uses:
* `learnmore` is independant, it could as well be in another file, we don't really care.
* `localizeShadowDom` depends on the actual element being localized, so we need the "this". (but see below)
* for `confirm`, because the handler is so small, I think it's more readable to have the handler inline.

Answering to @gabrielesvelto:

createdCallback is always called only once but now I realize from your comment that we don't free it when the element is removed. So I think I should attach on "attachedCallback" and detach on detachedCallback (we don't know when the object is garbage collected). But... we can't detach l10n's ready event. So I think I'll either have to use a global event handler and do my own attach/detach mechanism, or implement this in l10n.js.